File: php/features.php
<?php
declare(strict_types=1);

// Myip SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class MyipFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new MyipBaseFeature();
            case "test":
                return new MyipTestFeature();
            default:
                return new MyipBaseFeature();
        }
    }
}

File: py/myip_sdk/features.py
# Myip SDK feature factory

from myip_sdk.feature.base_feature import MyipBaseFeature
from myip_sdk.feature.test_feature import MyipTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MyipBaseFeature(),
        "test": lambda: MyipTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
